src/core/wappu-mood-core.js

```diff
@@ -0,0 +1,152 @@
+const {knex} = require('../util/database').connect();
+import {deepChangeKeyCase} from '../util';
+import * as feedCore from './feed-core';
+const requireEnvs = require('../util/require-envs');
+
+requireEnvs(['MOOD_START_DATE', 'MOOD_END_DATE']);
+
+function createOrUpdateMood(opts) {
+  const upsertMoodSql = `
+    WITH upsert AS
+      (UPDATE
+        wappu_mood
+      SET
+        rating = ?,
+        description = ?,
+        updated_at = CURRENT_TIMESTAMP
+      WHERE
+        user_id = ? AND created_at_coarse = CURRENT_DATE
+      RETURNING
+        *,
+        false AS is_new
+    ), inserted AS (
+        INSERT INTO
+          wappu_mood(user_id, rating, description)
+        SELECT ?, ?, ? WHERE NOT EXISTS( SELECT * FROM upsert )
+        RETURNING
+          *,
+          true AS is_new
+    )
+    SELECT *
+    FROM upsert
+    UNION ALL
+    SELECT *
+    FROM inserted;
+  `;
+
+  let params = [
+    opts.rating, opts.description, opts.client.id,
+    opts.client.id, opts.rating, opts.description
+  ];
+
+  return knex.transaction(trx =>
+    trx.raw(upsertMoodSql, params)
+      .then(result => {
+        const mood = result.rows[0];
+        if (mood.is_new) feedCore.createFeedItem(_feedTemplate(mood, opts.client));
```

Comment:
Missing curly brackets. Also the transaction needs to be passed to `createFeedItem` to do the feed item creation in the same transaction as mood update.